### ISSUE TITLE
Why doesn't amp improve training speed？

### ISSUE THREAD

**Issue by RichardcLee** [pytorch/pytorch#45420]
## 🐛 Bug

When I use amp, CUDA memory does save a lot, but the training speed is **much** lower!

## To Reproduce

1. My code: 
            
            `self.optimizer.zero_grad()
            with amp.autocast(enabled=self.opt.amp):
                # if deep sup: get multiple output(a tuple), else: get a batch(Tensor)
                output = self.model(src_img)    # forward

                lossT = self.loss_calculator.calc_loss(output, label, is_deep_sup=self.opt.deep_sup)

            # float16 + float32
            if self.opt.amp:
                self.scaler.scale(lossT).backward()   # Scales loss
                self.scaler.step(self.optimizer)
                self.scaler.update()   # Updates the scale for next iteration.

            # float32
            else:
                lossT.backward()
                self.optimizer.step()  # update params`

1. mini test:
use amp:
![image](https://user-images.githubusercontent.com/28611665/94409077-3e857800-01a8-11eb-96aa-873656e3bcc8.png)
not use:
![image](https://user-images.githubusercontent.com/28611665/94409101-48a77680-01a8-11eb-845f-f12175b63954.png)

## Environment

PyTorch version: 1.6.0
Is debug build: False
CUDA used to build PyTorch: 10.2
ROCM used to build PyTorch: N/A

OS: Microsoft Windows 10 
GCC version: Could not collect
Clang version: Could not collect
CMake version: Could not collect

Python version: 3.6 (64-bit runtime)
Is CUDA available: True
CUDA runtime version: 10.2.89
GPU models and configuration: Could not collect
Nvidia driver version: Could not collect
cuDNN version: Could not collect
HIP runtime version: N/A
MIOpen runtime version: N/A

Versions of relevant libraries:
[pip3] numpy==1.19.1
[pip3] torch==1.6.0
[pip3] torchsummary==1.5.1
[pip3] torchvision==0.7.0
[conda] blas                      1.0                         mkl    defaults
[conda] cudatoolkit               10.2.89              h74a9793_1    defaults
[conda] mkl                       2020.2                      256    defaults
[conda] mkl-service               2.3.0            py36hb782905_0    defaults
[conda] mkl_fft                   1.2.0            py36h45dec08_0    defaults
[conda] mkl_random                1.1.1            py36h47e9c7a_0    defaults
[conda] numpy                     1.19.1           py36h5510c5b_0    defaults
[conda] numpy-base                1.19.1           py36ha3acd2a_0    defaults
[conda] pytorch                   1.6.0           py3.6_cuda102_cudnn7_0    pytorch
[conda] torchsummary              1.5.1                    pypi_0    pypi
[conda] torchvision               0.7.0                py36_cu102    pytorch

cc @mcarilli

**Comment by mcarilli** [comment]
See
https://pytorch.org/tutorials/recipes/recipes/amp_recipe.html#speedup-with-amp-is-minor
https://pytorch.org/tutorials/recipes/recipes/tuning_guide.html
for general and amp guidance.

If your model/batch size are small and achieve low device utilization, amp won't help, and may hurt.  Also, if your GPU is not Tensor Core enabled (in other words if it's Pascal or earlier) amp isn't expected to deliver much speedup.  What model GPU are you using?

UNet has 3D convolutions (right?) so it's also possible you're hitting bad cudnn heuristics.  `torch.backends.cudnn.benchmark = True` at the top of your script may help.  If you find a particular layer/conv shape that performs poorly with FP16 even after `benchmark = True` I will report it to cudnn devs.

**Comment by RichardcLee** [comment]
Thank you for your patience.
Our model is 2D AttUnet(batch_size=8) and GPU is 1070Ti(pascal)
and we have already used torch.backends.cudnn.benchmark = True at the top of script, but it doesn't work.
So the reason for the problem should be the GPU.
